pkg/controller/defaults/placements.go

```diff
@@ -66,7 +66,8 @@ var (
 				getOcsToleration(),
 			},
 			TopologySpreadConstraints: []corev1.TopologySpreadConstraint{
-				getTopologySpreadConstraintsSpec(1, "rook-ceph-osd-prepare"),
+				getTopologySpreadConstraintsSpec(1, "rook-ceph-osd-prepare", "rook-ceph-osd"),
+				getTopologySpreadConstraintsSpec(1, "rook-ceph-osd-prepare", "rook-ceph-osd"),
```

Comment:
I think the reverse approach would be easier to understand - add the extra TSC entry and set it to zone/rack at runtime only if flexible scaling is disabled.

Reply:
Agree with @nbalacha 
This looks like a mess to me.

Reply:
+1

Reply:
Thanks,required changes have been done!